base/sharedarray.jl

```diff
@@ -307,21 +307,25 @@ localindexes(S::SharedArray) = S.pidx > 0 ? range_1dim(S, S.pidx) : 1:0
 unsafe_convert{T}(::Type{Ptr{T}}, S::SharedArray) = unsafe_convert(Ptr{T}, sdata(S))
 
 function convert(::Type{SharedArray}, A::Array)
-    S = SharedArray(eltype(A), size(A))
+    sz = size(A)
+    S = SharedArray{eltype(A),length(sz)}(sz)
```

Comment:
Note that there isn't any methods defined for `SharedArray{T,N}` apart from the inner constructor so this won't work. You need to either not use them and use `SharedArray{T}` instead or check for conflict/ambiguity and define them.